### PR TITLE
feat: implement database connection

### PR DESCRIPTION
**Pull request by joseluis-teran-jalafund**
## Trello Ticket

[ID-28](https://trello.com/c/Sb7ZnKnu/37-id-28-implement-data-and-database)

## What did I do?

I implemented the Unit of Work design pattern using Dapper for transactional database operations. This included creating a `NpgsqlConnectionFactory` to manage PostgreSQL connections, setting up repositories, and coordinating them through the Unit of Work to ensure consistent database operations.

## Why did I do this?

The goal was to ensure transactional consistency across multiple operations that may affect multiple repositories. By implementing the Unit of Work pattern, I was able to group these operations within a single transaction, ensuring that either all changes are committed or none at all. This pattern improves data integrity and performance by managing database connections more efficiently.

## How did I do it?

1. I created an interface `IDbConnectionFactory` to manage PostgreSQL connections and transactions asynchronously.
2. I built a `UnitOfWork` class that coordinates the repositories and manages transaction commit/rollback, ensuring that all changes are saved or discarded atomically.
3. The service layer interacts with the `UnitOfWork`, handling the business logic, and calling `CommitAsync` or `RollbackAsync` based on success or failure.

## (Optional) Notes:

- I used the `IDbConnectionFactory` to ensure that connections and transactions are managed centrally, and every repository shares the same connection context within a unit of work.
- I had to adjust the original implementation because IDbTransaction in .NET does not have CommitAsync and RollbackAsync, so I used the synchronous methods and returned a Task.CompletedTask to maintain the asynchronous flow.
- This setup uses the Dapper ORM

### CODE REVIEW

File: src/ExpenseTrackerGrupo2/Persistence/Database/DbInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace ExpenseTrackerGrupo2.Persistence.Database;
+
+public class DbInitializer
+{
+    // TODO: Complete according the sql scripts
```
Comment (alex-paca-jalafund) [inline src/ExpenseTrackerGrupo2/Persistence/Database/DbInitializer.cs:5]: We set the database through docker-compose. I dont know if is still needed. What do you think?

**Comment by gabriela-garcia-jalafund** [comment]
@joseluis-teran-jalafund I have no observations other than what my colleague mentioned to me. @alex-paca-jalafund

**Review by gabriela-garcia-jalafund — APPROVED** [review]

**Review by alex-paca-jalafund — APPROVED** [review]
We need this changes,  to make the fixes, due this PR has part of the DB connections.

**Comment by alex-paca-jalafund** [comment]
[ID-30: Resolve DB Connection](https://trello.com/c/XQE1iyNu/41-id-30-resolve-db-connection)

**Pull request merged**